File: backend/app/schemas/weather.py
from pydantic import BaseModel


class WeatherResult(BaseModel):
    condition: str  # OpenWeather's "main" bucket, e.g. Clear/Clouds/Rain/Thunderstorm/Snow/Fog
    description: str | None = None
    temp_c: float | None = None
    feels_like_c: float | None = None
    humidity: int | None = None
    wind_speed_ms: float | None = None
    icon: str | None = None  # OpenWeather icon code (e.g. "01d")
    location_name: str | None = None

File: frontend/src/api/weather.ts
import { apiClient } from './client';

export interface WeatherData {
  condition: string;
  description: string | null;
  temp_c: number | null;
  feels_like_c: number | null;
  humidity: number | null;
  wind_speed_ms: number | null;
  icon: string | null;
  location_name: string | null;
}

export type CurrentWeather = WeatherData;

export async function fetchCurrentWeather(): Promise<WeatherData> {
  const { data } = await apiClient.get<WeatherData>('/weather/current');
  return data;
}